tests/Rule/MultiplyAndDivideByStringRector/Fixture/skip_union_types.php.inc

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Rector\Money\Tests\Rule\MultiplyAndDivideByStringRector\Fixture;
+
+use Money\Currency;
+use Money\Money;
+use function mt_rand;
+
+function returnInt(): int
+{
+    return 1;
+}
+
+class UnionReturning
+{
+    /**
+     * @return Money|array|false
+     */
+    public static function foo()
+    {
+        $random = mt_rand(0, 10);
+        if ($random % 3 === 0) {
+            return new Money($random, new Currency('EUR'));
+        }
+        if ($random === 5) {
+            return ['some' => 'other', 'data'];
+        }
+        return false;
+    }
+
+    /**
+     * @return Money|array|false
+     */
+    public function bar()
+    {
+        return self::foo();
+    }
+}
+
+$unionReturning = new UnionReturning();
+$unionReturning->bar()->divide(1);
+$unionReturning->bar()->multiply(2);
+
+UnionReturning::foo()->divide(3);
+UnionReturning::foo()->multiply(4);
```

Comment:
Please use floats in all `divide()` / `multiply()` calls, since integers were already ignored as valid usage. Only floats are converted to strings, so in order to see if union types are actually ignored, we need to call those methods with float 🙂

Reply:
I have added some cases, nonetheless as long as I don't add some logic that checks the union against containing Money etc. this will have no impact (see diff)

Reply:
@pascalheidmann FYI: PHPStan failures are not related to your changes, the same errors can be reproduced on `main` branch after updating dependencies. [It's most probably related to Rector >=0.13.7](https://twitter.com/_Codito_/status/1551696130735480832). I'll try to fix them and rebase your branch after that.